src/compose/ComposeBox.js

```diff
@@ -217,6 +220,47 @@ class ComposeBox extends PureComponent<Props, State> {
     }
   };
 
+  insertAttachment = async (attachments: DocumentPickerResponse[]) => {
+    this.setState(({ numUploading }) => ({
+      numUploading: numUploading + 1,
+    }));
+
+    const { _, auth } = this.props;
+
+    let placeholders = '';
+    for (let i = 0; i < attachments.length; i++) {
+      const fileName = attachments[i].name ?? `attachment-${i + 1}`;
+      placeholders += `${_('[Uploading {fileName}...]()', { fileName })}\n\n`;
```

Comment:
this counts as a string concatenation which is denied in [/docs/howto/translations.md](https://github.com/zulip/zulip-mobile/blob/8430b6da8/docs/howto/translations.md)

> Never try to concatenate translated strings together, or do other string manipulation on them.

is this reasonable code? I kept it here since it only involves concatenation with newlines, which shouldn't create a problem according to me.

Reply:
Thanks for asking. Yeah, this is fine as far as that's concerned -- what that's really meant to be about is attempting to put together a sentence with translated text, because the structure of the sentence will be different in different languages. I guess I should try to clarify that bit in the doc.

Reply:
4bb2a4a78